testcases/OpTestGSBStaticKey.py

```diff
@@ -0,0 +1,308 @@
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2023
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+'''
+OpTestGSBStaticKey
+------------
+Test Enable and Disable Secure boot
+
+Steps:
+Grub and kernel files should be signed with appended signatures
+Enable/Disable Secure boot at Operating System level
+Shutdown LPAR
+Enable/Disable Secure boot at HMC level
+Activate LPAR
+'''
+import unittest
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+
+
+class OpTestGSBStaticKey(unittest.TestCase):
+    """
+    Class OpTestGSBStaticKey:
+    """
+    def setUp(self):
+        # Connection variables
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.cv_HMC = self.cv_SYSTEM.hmc
+        self.c = self.cv_HMC.get_host_console()
+        self.hmc_con = self.cv_HMC.ssh
+        # Variables required for this test
+        self.hmc_secureboot = False
+        self.os_secureboot = False
+        self.dt_secureboot = False
+        self.lockdown = False
+        self.sys_lockdown = ""
+        self.grubFilename = ""
+        self.prepDisk = ""
+        self.signature = "~Module signature appended~"
+
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        os_level = self.cv_HOST.host_get_OS_Level()
+        if 'Red Hat' in os_level:
+            self.distro = 'rhel'
+        elif 'SLES' in os_level:
+            self.distro = 'sles'
+        else:
+            raise self.skipTest("Test currently supported on "
+                                "SLES and RHEL releases")
+
+    def check_hmc_secureboot_state(self):
+        '''
+        Assign 'self.hmc_secureboot' with
+        'True' in case of Secure boot enabled
+        'False' in case of Secure boot disabled
+        '''
+        # HMC command to know the current state of Secure Boot
+        cmd = ("lssyscfg -r lpar -m %s -F curr_secure_boot --filter "
+               "lpar_names=%s" %
+               (self.cv_HMC.mg_system, self.cv_HMC.lpar_name))
+        output = self.hmc_con.run_command(cmd, timeout=300)
+        if int(output[0]) == 2: # Value '2' means Secure Boot enabled
+            self.hmc_secureboot = True
+        elif int(output[0]) == 0: # Value '0' means Secure Boot disabled
+            self.hmc_secureboot = False
+
+    def hmc_secureboot_on_off(self, enable=True):
+        '''
```

Comment:
better to have check_hmc_secureboot_state and  hmc_secureboot_on_off into util in OptestHMC.py .. in case if other test needs to run with secure boot enable

Reply:
yes seems me a good idea but we can pursue this as an enhancement

Reply:
This function depends on other functions and not self-sufficient to move in to OptestHMC.py. So not doing this change.